src/runtime/graph_executor/debug/graph_executor_debug.cc

```diff
@@ -208,25 +201,70 @@ class GraphExecutorDebug : public GraphExecutor {
   }
 
   /*!
-   * \brief Copy index-th node to data_out.
+   * \brief Get number of outputs of the node.
+   * \param index The index of the node.
+   * \return The number of outputs.
+   */
+  size_t GetNodeNumOfOutputs(size_t index) {
+    if (nodes_[index].op_type != "tvm_op") return 1;
+    return static_cast<size_t>(nodes_[index].param.num_outputs);
+  }
+
+  /*!
+   * \brief Execute network to a specific node and return result.
    *
    * This method will do a partial run of the the graph
    * from begining upto the index-th node and return output of index-th node.
    * This is costly operation and suggest to use only for debug porpose.
    *
-   * \param index: The  index of the node.
-   * \param data_out the node data.
+   * \param index: The index of the node.
+   * \param eid The Entry id of the op.
+   * \return Node output array.
    */
-  void DebugGetNodeOutput(int index, DLTensor* data_out) {
+  NDArray ExecuteNodeGetOutput(int index, int eid) {
     ICHECK_LT(static_cast<size_t>(index), op_execs_.size());
-    uint32_t eid = index;
 
-    for (size_t i = 0; i < op_execs_.size(); ++i) {
+    for (size_t i = 0; i <= static_cast<size_t>(index); i++) {
       if (op_execs_[i]) op_execs_[i]();
-      if (static_cast<int>(i) == index) break;
     }
 
-    data_entry_[eid].CopyTo(data_out);
+    return data_entry_[entry_id(index, eid)];
+  }
+
+  /*!
+   * \brief Execute next node in the network.
+   *
+   * This method will execute next node assuming
+   * previous nodes has been executed and return output of index-th node.
+   *
+   * \param index: The index of the node.
+   * \param eid The Entry id of the op.
+   * \return Node output array.
+   */
+  NDArray ExecuteNextNodeGetOutput(int index, int eid) {
+    ICHECK_LT(static_cast<size_t>(index), op_execs_.size());
+    if (op_execs_[index]) op_execs_[index]();
+    return data_entry_[entry_id(index, eid)];
+  }
+
+  /*!
+   * \brief Execute full network and store each layer outputs.
+   *
+   * This method will execute a full network and store layer
+   * outputs along execution.
+   *
+   * \return All node outputs.
+   */
+  Array<NDArray> RunAndGetLayersOutputs() {
```

Comment:
This function should return an `Array<Array<NDArray>>` where the outer node corresponds to the layer and the inner array is that layers outputs.